src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -798,7 +798,10 @@ def setter(webapp):
 
 
 def show_identity(cmd, resource_group_name, name, slot=None):
-    return _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot).identity
+    op = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot)
+    if not op:
+        raise ValidationError("No identity to show for function app")
```

Comment:
this method is common to function apps & WebApps. Update the message to something like "Unable to find App {}  in resource group {} "